### ISSUE TITLE
How do set GPU memory of 06_classify_image.py

### ISSUE THREAD

**Issue by daniellinboy** [jensleitloff/CNN-Sentinel#7]
My computer GPU memory have 16G.
GPU memory use about 16G when train_rgb_finetuping.py, but run classify image.py only use  about 110 MB.
I don't know how do set memory value in  06_classify_image.py.

Thanks.

**Comment by felixriese** [comment]
**Tl;dr:** It sounds as you are fine and everything runs as expected.

The training in [02_train_rgb_finetuning.py](https://github.com/jensleitloff/CNN-Sentinel/blob/master/py/02_train_rgb_finetuning.py) needs a lot of resources and if the GPU of your computer is used by the script, you are fine. This is the case as far as I understand you, @daniellinboy.

The classification (meaning: prediction) in [06_classify_image.py](https://github.com/jensleitloff/CNN-Sentinel/blob/master/py/06_classify_image.py) does not need as many GPU resources but more RAM and more CPU.

**Comment by daniellinboy** [comment]
@felixriese 
I know you mean.
Thanks very much for the assistance you provide my help.

More further,I obtain result of S2 image. But I don't know value range 0.67-0.83 of ms_prob why not get an integer value. [attachment figure](https://drive.google.com/file/d/1qdg4pz7BbdjdGPhtFXLGtJi6QHm5xILn/view?usp=sharing)

Sorry to bother you.

**Comment by jensleitloff** [comment]
**Tl;dr**: Memory reservation is not memory usage

Just an additional comment on the initial topic:
Tensorflow is greedy in memory consumption. In general, it reserves (but not necessarily uses) all available memory of all available GPUs. Thus, you will get the impression, that during training your 16GB are in use. I strongly assume, that this is not true, as the training is also possible on much less VRAM. 

I can only guess the reasons, why the classification uses only 110 MB in your case. But I suspect, that either another process is blocking VRAM and thus the classification can only reserve this small amount or the classification is not running on the GPU at all (please check both with `nvidia-smi -l 1`)
There is a good blog entry on Tensorflow's GPU memory usage and how to monitor the true amount of used (and not only reserved) VRAM: [TensorFlow’s “Its Complicated” Relationship with GPU Memory](https://medium.com/@auro_227/tensorflows-its-complicated-relationship-with-gpu-memory-5672745df84)

Finally, I just want to mention, that our provided code is for demonstration. It's far from highly optimized.

**Comment by daniellinboy** [comment]
Thank you for the information.
I just guessed to train in tensorflow-gpu (conda environment) when running VRAM successfully. But when VRAM uses a small amount, I will actively use pycon to classify the image. Now, I am trying hard.
